src/panel/direction/DirectionPanel.jsx

```diff
@@ -56,11 +59,23 @@ export default class DirectionPanel extends React.Component {
     this.restorePoints(props);
   }
 
-  componentDidMount() {
+  async componentDidMount() {
     Telemetry.add(Telemetry.ITINERARY_OPEN);
     document.body.classList.add('directions-open');
     this.dragPointHandler = listen('change_direction_point', this.changeDirectionPoint);
     this.setPointHandler = listen('set_direction_point', this.setDirectionPoint);
+
+    if (!this.props.origin && isMobileDevice()) {
+      // If authorized, set origin to current position on mobile
+      await GeolocationCheck.showGeolocationModalIfNeeded();
```

Comment:
Why do you need to do it here explicitely, whereas it's already done in the `geolocate` method?